stripe/src/main/java/com/stripe/android/view/CardInputWidget.java

```diff
@@ -937,37 +865,9 @@ private void updateIconForCvcEntry(boolean isAmEx) {
     }
 
     /**
-     * Class used to encapsulate the functionality of "backing up" via the delete/backspace key
-     * from one text field to the previous. We use this to simulate multiple fields being all part
-     * of the same EditText, so a delete key entry from field N+1 deletes the last character in
-     * field N. Each BackUpFieldDeleteListener is attached to the N+1 field, from which it gets
-     * its {@link #onDeleteEmpty()} call, and given a reference to the N field, upon which
-     * it will be acting.
-     */
-    private class BackUpFieldDeleteListener implements StripeEditText.DeleteEmptyListener {
-
-        private StripeEditText backUpTarget;
-
-        BackUpFieldDeleteListener(StripeEditText backUpTarget) {
-            this.backUpTarget = backUpTarget;
-        }
-
-        @Override
-        public void onDeleteEmpty() {
-            String fieldText = backUpTarget.getText().toString();
-            if (fieldText.length() > 1) {
-                backUpTarget.setText(
-                        fieldText.substring(0, fieldText.length() - 1));
-            }
-            backUpTarget.requestFocus();
-            backUpTarget.setSelection(backUpTarget.length());
-        }
-    }
-
-    /**
      * A data-dump class.
      */
-    class PlacementParameters {
+    static class PlacementParameters {
```

Comment:
It doesn't have any testing implications really - at least not how I was using it. Just avoiding the automatic implicit reference to the containing class and possible memory leak. I'm not sure whether or not I would have injected this with DI if we had some -- the purpose of this was really just to group all the placements together as a sort of "info dump" struct that had the convenient property of being easy to fake when testing (and the toString was super nice for debugging)